features/api_full_auto_attach.feature

```diff
@@ -0,0 +1,54 @@
+Feature: Full Auto-Attach Endpoint
+
+    @series.lts
+    @uses.config.machine_type.aws.pro
+    @uses.config.machine_type.azure.pro
+    @uses.config.machine_type.gcp.pro
+    Scenario Outline: Run auto-attach on cloud instance.
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        contract_url: 'https://contracts.canonical.com'
+        data_dir: /var/lib/ubuntu-advantage
+        log_level: debug
+        log_file: /var/log/ubuntu-advantage.log
+        """
+        When I create the file `/tmp/full_auto_attach.py` with the following:
+        """
+        from uaclient.api.u.pro.attach.auto.full_auto_attach.v1 import full_auto_attach, FullAutoAttachOptions
+
+        full_auto_attach(FullAutoAttachOptions(enable=["esm-infra", "esm-apps"]))
+        """
+        And I run `python3 /tmp/full_auto_attach.py` with sudo
+        Then stdout matches regexp:
+        """
+        Ubuntu Pro: ESM Apps enabled
+        """
+        And stdout matches regexp:
+        """
+        Ubuntu Pro: ESM Infra enabled
+        """
+        When I run `pro status --all` with sudo
+        Then stdout matches regexp:
+        """
+        SERVICE       +ENTITLED  STATUS    DESCRIPTION
+        cc-eal        +yes +<cc-eal-s>  +Common Criteria EAL2 Provisioning Packages
+        """
+        Then stdout matches regexp:
+        """
+        esm-apps      +yes +enabled +Extended Security Maintenance for Applications
+        esm-infra     +yes +enabled +Extended Security Maintenance for Infrastructure
+        fips          +yes +<fips-s> +NIST-certified core packages
+        fips-updates  +yes +<fips-s> +NIST-certified core packages with priority security updates
+        livepatch     +yes +enabled  +Canonical Livepatch service
+        """
+        Then stdout matches regexp:
+        """
+        <cis_or_usg>           +yes  +<cis-s>  +Security compliance and audit tools
+        """
```

Comment:
We only really need to test that
1. the machine is attached
2. the services that got passed in the `enable` param got enabled
3. the services that normally get enabled by default but aren't in the `enable` array did not get enabled
```suggestion
        When I run `pro status --all` with sudo
        Then stdout matches regexp:
        """
        esm-apps      +yes +enabled +Extended Security Maintenance for Applications
        esm-infra     +yes +enabled +Extended Security Maintenance for Infrastructure
        """
        Then stdout matches regexp:
        """
        livepatch     +yes +disabled  +Canonical Livepatch service
        """
```

Reply:
Oh yes,